lib/compilation-env.ts

```diff
@@ -41,6 +41,8 @@ import type {PropertyGetter} from './properties.interfaces.js';
 import {CompilerProps} from './properties.js';
 import {createStatsNoter, IStatsNoter} from './stats.js';
 
+type PropFunc = (string, any?) => any;
```

Comment:
Can you not use the existing `PropertyGetter` here?

Reply:
see explanation about the `_.partial()` thing